src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator, InfoImplementer {
+    private String color = new ColorSupplier().getRandomColor();
+
+    private String name;
+
+    private float something = new FigureSupplier().getRandomFigure();
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public float getArea() {
+        return 0;
+    }
+
+    @Override
+    public StringBuilder printInfo() {
+        return null;
+    }
```

Comment:
abstract class doesn't have to implement all methods
let's remove these methods